main.go

```diff
@@ -173,15 +193,15 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 }
 
 func main() {
-	runtime.GOMAXPROCS(2)
-	if len(os.Args) != 2 {
-		log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
-	}
-
+	runtime.GOMAXPROCS(5)
+	cert := flag.String("cert", "", "certificate file")
+	key := flag.String("key", "", "key file")
+	flag.Parse()
 	mfChan := make(chan *dto.MetricFamily, 1024)
-
-	go fetchMetricFamilies(os.Args[1], mfChan)
-
+	if (*cert != "" && *key == "") || (*cert == "" && *key != "") {
```

Comment:
Sorry, one more possible change to make: the original check to see if there is no metrics URL argument is lost, it appears.  Can we add back a check like:
```
if len(flag.Args()) != 1 {
    log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
}
```

Possible?